spec/rubocop/cop/rspec/predicate_matcher_spec.rb

```diff
@@ -312,6 +326,23 @@
         RUBY
       end
 
+      it 'does not register an offense for a predicate method ' \
+         'with multiple arguments and not replaceable method' do
+        expect_no_offenses(<<~RUBY)
+          expect(foo).to be_a(foo, bar)
+          expect(foo).to be_an foo, bar
+          expect(foo).to be_a_kind_of(foo, bar, baz)
+          expect(foo).to a_kind_of foo, bar, baz
+          expect(foo).to be_kind_of(foo, bar)
+          expect(foo).to be_an_instance_of foo, bar
+          expect(foo).to be_instance_of(foo, bar, baz)
+          expect(foo).to an_instance_of foo, bar, baz
+          expect(foo).to include(foo, bar)
+          expect(foo).to respond_to foo, bar
+          expect(foo).to have_key(foo, bar, baz)
```

Comment:
Is this even possible?

I suggest to only target those predicate matchers in this fix that can reasonably accept more than one argument.

`include` is a special case, as it's behaves differently depending on the `actual`'s type. And it [accepts varargs](https://github.com/rspec/rspec-expectations/blob/4ef5b0e21c2273a14ff46b2382ab921e7b100d72/lib/rspec/matchers/built_in/include.rb#L15).

Are all other listed predicate matchers like that?

Reply:
I thought there might be a case where it would be a false positive if the method was overridden.
But is it excessive to care so much?

Reply:
If it's overridden - it won't result in an ArgumentError, would it?

Reply:
It is likely that the method to be replaced must be overridden in the same way. So, I am trying to avoid unnecessary autocorrects and offense here. 

So I think it would be better not to make it an offense here except for methods that can exist without override.

Reply:
`be_a` [won't accept two arguments](https://github.com/rspec/rspec-expectations/blob/4ef5b0e21c2273a14ff46b2382ab921e7b100d72/lib/rspec/matchers.rb#L355):
```
> be_a(Integer, 1).matches?(1)
ArgumentError: wrong number of arguments (given 2, expected 1)
from /Users/pirj/.rvm/gems/ruby-3.0.3@rubocop-rspec/gems/rspec-expectations-3.12.1/lib/rspec/matchers.rb:355:in `be_a'
```

How would the original `be_a(foo, bar)` code even work?

Reply:
> be_a [won't accept two arguments](https://github.com/rspec/rspec-expectations/blob/4ef5b0e21c2273a14ff46b2382ab921e7b100d72/lib/rspec/matchers.rb#L355):

Yes. I understand about that.

>How would the original be_a(foo, bar) code even work?

In the case of the original, I believe it would be an ArgumentError as you say.

So I didn't know if it was a good thing to give it as a violation since it could be a custom matcher, etc., redefined by the user.
The reason is that the methods to which they are substituted must also be overridden and extended to multiple arguments.

However, I don't have a use case to extend these either, so I will not consider these. Thank you for taking this up.